test/python/synthesis/test_mcx_synthesis.py

```diff
@@ -0,0 +1,123 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""Test Multi Controlled X Gate synthesis functions."""
+
+import unittest
+import numpy as np
+from ddt import ddt, data
+
+from qiskit import QuantumCircuit, transpile
+from qiskit.quantum_info.states import Statevector
+from qiskit.synthesis.multi_controlled import (
+    synth_mcx_1_clean_kg24,
+    synth_mcx_2_clean_kg24,
+    synth_mcx_1_dirty_kg24,
+    synth_mcx_2_dirty_kg24,
+)
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+
+@ddt
+class TestMCXSynth(QiskitTestCase):
```

Comment:
Nothing to do right now, but just a reminder to merge this with the tests added in #13961 once that's merged

Reply:
Rebased onto #13961